tensorboard/plugins/text_v2/text_v2_plugin_test.py

```diff
@@ -0,0 +1,230 @@
+            "truncated"
+        ]
+        self.assertEqual(truncated, False)
+
+    def test_reduce_to_2d(self):
+        def make_range_array(dim):
+            """Produce an incrementally increasing multidimensional array.
+
+            Args:
+              dim: the number of dimensions for the array
+
+            Returns:
+              An array of increasing integer elements, with dim dimensions and size
+              two in each dimension.
+
+            Example: rangeArray(2) results in [[0,1],[2,3]].
+            """
+            return np.array(range(2 ** dim)).reshape([2] * dim)
+
+        for i in range(2, 5):
+            actual = text_v2_plugin.reduce_to_2d(make_range_array(i))
+            expected = make_range_array(2)
+            np.testing.assert_array_equal(actual, expected)
+
+    def test_reduce_and_jsonify(self):
+        convert = text_v2_plugin.reduce_and_jsonify
+        scalar = np.array("foo")
+        scalar_expected = ("foo", (), False)
+        self.assertEqual(convert(scalar), scalar_expected)
+
+        # Check that underscores are preserved correctly; this detects erroneous
+        # use of UTF-16 or UTF-32 encoding when calling markdown_to_safe_html(),
+        # which would introduce spurious null bytes and cause undesired <em> tags
+        # around the underscores.
+        scalar_underscores = np.array("word_with_underscores")
+        scalar_underscores_expected = ("word_with_underscores", (), False)
+        self.assertEqual(
+            convert(scalar_underscores), scalar_underscores_expected
+        )
+
+        vector = np.array(["foo", "bar"])
+        vector_expected = (["foo", "bar"], (2,), False)
+        self.assertEqual(convert(vector), vector_expected)
+
+        d2 = np.array([["foo", "bar"], ["zoink", "zod"]])
+        d2_expected = ([["foo", "bar"], ["zoink", "zod"]], (2, 2), False)
+        self.assertEqual(convert(d2), d2_expected)
+
+        d3 = np.array(
+            [
+                [["foo", "bar"], ["zoink", "zod"]],
+                [["FOO", "BAR"], ["ZOINK", "ZOD"]],
+            ]
+        )
+
+        d3_expected = ([["foo", "bar"], ["zoink", "zod"]], (2, 2, 2), True)
+        self.assertEqual(convert(d3), d3_expected)
+
+    @with_plugin(generate_testdata=False)
+    def testPluginIsActiveWhenNoRuns(self, plugin):
+        """The plugin should be inactive when there are no runs."""
+        self.assertFalse(plugin.is_active())
+
+    @with_plugin()
+    def testPluginIsActiveWhenTextRuns(self, plugin):
+        """The plugin should be active when there are runs with text."""
+        self.assertFalse(plugin.is_active())
+
+    @with_plugin(include_text=False)
+    def testPluginIsActiveWhenRunsButNoText(self, plugin):
+        """The plugin should be inactive when there are runs but none has
+        text."""
+        self.assertFalse(plugin.is_active())
+
+    @with_plugin()
+    def testPluginIndexImpl(self, plugin):
```

Comment:
This test looks to be identical to `testIndex()` - I think it can just be deleted.

Reply:
done